analysis.go

```diff
@@ -155,9 +156,18 @@ func ListPackages(fileRoot, importRoot string) (PackageTree, error) {
 		ip := filepath.ToSlash(filepath.Join(importRoot, strings.TrimPrefix(path, fileRoot)))
 
 		// Find all the imports, across all os/arch combos
-		p, err := ctx.ImportDir(path, analysisImportMode())
+		ap, err := filepath.Abs(path)
```

Comment:
oh wait i think i know - wanting to make sure that there's no confusion about what goes in the `pkgcache`.

so, that cache is actually a problem, heh - this same path may well get used later on for a different version of the same code. when we need to inspect different versions of the code for their characteristics, we swap them out in-place.

there's caches in place higher up, though, so dropping that cache would be fine.